docs/logs/complex-objects/Program.cs

```diff
@@ -34,5 +34,5 @@ internal static partial class LoggerExtensions
     [LoggerMessage(LogLevel.Critical)]
     public static partial void FoodRecallNotice(
         this ILogger logger,
-        [LogProperties(OmitReferenceName = true)] FoodRecallNotice foodRecallNotice);
+        [LogProperties(OmitReferenceName = true)] in FoodRecallNotice foodRecallNotice);
```

Comment:
`in` with mutable structs can be tricky. If the compiler can't determine stuff is `readonly` it will make defensive copies. I think the auto-properties shown here are fine (IIRC the getters get marked as `readonly` automatically for auto-properties), but users could run into trouble if they handroll more. If we want to recommend `in` we may also want to recommend the `struct`s be `readonly` too? `readonly struct` + `in` work always.

Reply:
Build failed, seems it is only available in .NET Core. I'll revert the change.